### PR TITLE
Fix error message being shown for duplicated elements

### PR DESCRIPTION
**Pull request by gayathrir11**
## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

error

### CODE REVIEW

**Review by MauricioUyaguari — COMMENTED** [review]
small comment. Thanks!

File: packages/legend-graph/src/graph/MetaModelConst.ts [hunk 1]
```diff
@@ -42,6 +42,14 @@ export enum ROOT_PACKAGE_NAME {
   PROJECT_DEPENDENCY_ROOT = 'PROJECT_DEPENDENCY_ROOT',
 }
 
+export enum ELEMENT_ORIGIN {
+  CORE_MODELS = 'core models',
+  CURRENT_GRAPH = 'current graph',
```
Comment (akphi) [inline packages/legend-graph/src/graph/MetaModelConst.ts:47]: for current graph, we probably don't need to mention anything

File: packages/legend-graph/src/graph/PureModel.ts [hunk 2]
```diff
@@ -581,6 +582,40 @@ export class PureModel extends BasicModel {
     return element;
   }
 
+  getElementOrigin(path: string, includePakcage?: boolean): string | undefined {
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-graph/src/graph/PureModel.ts:585]: Also, I think these are a little expensive, should we call methods like `isSystemElement`, `isGeneratedElement`, etc.?

File: packages/legend-graph/src/graph-manager/__tests__/buildGraph/GraphBuildFailure.test.ts [hunk 3]
```diff
@@ -74,7 +74,9 @@ test(unitTest('Duplicated element'), async () => {
       TEST_DATA__DuplicatedElement as Entity[],
       graphManagerState.graphBuildState,
     ),
-  ).rejects.toThrowError(`Element 'ui::test1::Animal' already exists`);
+  ).rejects.toThrowError(
+    `Current graph element 'ui::test1::Animal' already exists in current graph`,
```
Comment (akphi) [inline packages/legend-graph/src/graph-manager/__tests__/buildGraph/GraphBuildFailure.test.ts:78]:
```suggestion
    `Element 'ui::test1::Animal' already exists`,
```

File: packages/legend-graph/src/graph-manager/__tests__/buildGraph/Dependency.test.ts [hunk 4]
```diff
@@ -181,6 +181,8 @@ test(
         firstDependencyEntities.entities,
         graphManagerState.graphBuildState,
       ),
-    ).rejects.toThrowError(`Element 'model::ClassB' already exists`);
+    ).rejects.toThrowError(
+      `Current graph element 'model::ClassB' already exists in dependency group-1:artifact-1`,
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-graph/src/graph-manager/__tests__/buildGraph/Dependency.test.ts:212]:
```suggestion
      `Element 'model::ClassB' already exists in dependency group-1:artifact-1`,
```

File: packages/legend-graph/src/graph/MetaModelConst.ts [hunk 5]
```diff
@@ -42,6 +42,14 @@ export enum ROOT_PACKAGE_NAME {
   PROJECT_DEPENDENCY_ROOT = 'PROJECT_DEPENDENCY_ROOT',
 }
 
+export enum ELEMENT_ORIGIN {
```
Comment (akphi) [inline packages/legend-graph/src/graph/MetaModelConst.ts:45]: can we maybe use `ROOT_PACKAGE_NAME` instead to not have to create yet another categorization that is semantically equivalent?

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-graph/src/graph/PureModel.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gayathrir11) [reply packages/legend-graph/src/graph/PureModel.ts:585]: I don't think we have that file anymore.

[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-graph/src/graph/PureModel.ts]: so let's do `DomainHelper` I guess

File: packages/legend-graph/src/graph/PureModel.ts [hunk 6]
```diff
@@ -581,6 +587,34 @@ export class PureModel extends BasicModel {
     return element;
   }
 
+  getElementOrigin(
```
Comment (akphi) [inline packages/legend-graph/src/graph/PureModel.ts:590]: this and the one in dependency manager's method signature make little sense, let's just do `getElementOrigin(element: PackageableElement): string | undefined`

File: packages/legend-graph/src/graph/PureModel.ts [hunk 7]
```diff
@@ -581,6 +587,34 @@ export class PureModel extends BasicModel {
...
+    let packageableElement: PackageableElement;
+    if (element) {
+      packageableElement = element;
+    } else {
+      packageableElement = guaranteeNonNullable(
+        this.getNullableElement(elementPath),
+      );
+    }
+    if (isSystemElement(packageableElement)) {
+      _origin = ROOT_PACKAGE_NAME.SYSTEM.slice(0, -5);
```
Comment (akphi) [inline packages/legend-graph/src/graph/PureModel.ts:604]: this looks really bizzare, why don't we just hard-code text here, after all, this method is basically a `pretty print` of element origin.

**Review by akphi — CHANGES_REQUESTED** [review]

**Pull request merged**